src/thor/costmatrix.cc

```diff
@@ -286,40 +288,31 @@ void CostMatrix::SourceToTarget(Api& request,
                                                                     .edgeid(),
                                                                 tile),
                                 time);
-      auto* pbf_date_time = matrix.mutable_date_times()->Add();
-      *pbf_date_time = dt_info.date_time;
-
-      auto* pbf_time_zone_offset = matrix.mutable_time_zone_offsets()->Add();
-      *pbf_time_zone_offset = dt_info.time_zone_offset;
-
-      auto* pbf_time_zone_name = matrix.mutable_time_zone_names()->Add();
-      *pbf_time_zone_name = dt_info.time_zone_name;
+      *matrix.mutable_date_times(connection_idx) = dt_info.date_time;
+      *matrix.mutable_time_zone_offsets(connection_idx) = dt_info.time_zone_offset;
+      *matrix.mutable_time_zone_names(connection_idx) = dt_info.time_zone_name;
     } else {
-      // Add empty strings to make sure pbf arrays are populated (serializer
-      // requires this)
-      auto* pbf_date_time = matrix.mutable_date_times()->Add();
-      *pbf_date_time = "";
-      auto* pbf_time_zone_offset = matrix.mutable_time_zone_offsets()->Add();
-      *pbf_time_zone_offset = "";
-      auto* pbf_time_zone_name = matrix.mutable_time_zone_names()->Add();
-      *pbf_time_zone_name = "";
+      // let's try a second pass for this connection
+      matrix.mutable_second_pass()->Set(connection_idx, true);
```

Comment:
`second_pass` is defaulting to false, `true` is signaling the second pass that it wasn't found initially